swift-sdk/Internal/InAppManager.swift

```diff
@@ -18,46 +26,121 @@ class InAppManager : IterableInAppManagerProtocol {
          inAppDelegate: IterableInAppDelegate,
          urlDelegate: IterableURLDelegate?,
          customActionDelegate: IterableCustomActionDelegate?,
-         urlOpener: UrlOpenerProtocol) {
+         urlOpener: UrlOpenerProtocol,
+         applicationStateProvider: ApplicationStateProviderProtocol,
+         notificationCenter: NotificationCenterProtocol,
+         dateProvider: DateProviderProtocol,
+         retryInterval: Double) {
+        ITBInfo()
         self.synchronizer = synchronizer
         self.displayer = displayer
         self.inAppDelegate = inAppDelegate
         self.urlDelegate = urlDelegate
         self.customActionDelegate = customActionDelegate
         self.urlOpener = urlOpener
+        self.applicationStateProvider = applicationStateProvider
+        self.notificationCenter = notificationCenter
+        self.dateProvider = dateProvider
+        self.retryInterval = retryInterval
+        
+        super.init()
         
         self.synchronizer.inAppSyncDelegate = self
+
+        self.notificationCenter.addObserver(self,
+                                       selector: #selector(onAppEnteredForeground(notification:)),
+                                       name: Notification.Name.UIApplicationDidBecomeActive,
+                                       object: nil)
+    }
+    
+    deinit {
+        ITBInfo()
+        notificationCenter.removeObserver(self)
     }
     
     func getMessages() -> [IterableInAppMessage] {
         ITBInfo()
-        return Array(messagesMap.values)
+        return Array(messagesMap.values.filter { $0.consumed == false })
```

Comment:
This list should keep the order as it was in the remote queue (and new messages should be added to the end of the queue).